crates/ruff_linter/src/rules/airflow/rules/deprecated_members.rs

```diff
@@ -0,0 +1,92 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::Expr;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+#[derive(Debug, Eq, PartialEq)]
+enum Replacement {
+    None,
+    Name(String),
+}
+
+impl Replacement {
+    fn name(name: impl Into<String>) -> Self {
+        Self::Name(name.into())
+    }
+}
+
+/// ## What it does
+/// Checks for uses of deprecated Airflow functions and values.
+///
+/// ## Why is this bad?
+/// Airflow 3.0 removed various deprecated functions, members, and other
+/// values. Some have more modern replacements. Others are considered too niche
+/// and not worth to be maintained in Airflow.
+///
+/// ## Example
+/// ```python
+/// from airflow.utils.dates import days_ago
+///
+///
+/// yesterday = days_ago(today, 1)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from datetime import timedelta
+///
+///
+/// yesterday = today - timedelta(days=1)
+/// ```
+#[violation]
+pub struct AirflowDeprecatedMembers {
+    deprecated: String,
+    replacement: Replacement,
+}
+
+impl Violation for AirflowDeprecatedMembers {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let AirflowDeprecatedMembers {
+            deprecated,
+            replacement,
+        } = self;
+        match replacement {
+            Replacement::None => format!("`{deprecated}` is removed in Airflow 3.0"),
+            Replacement::Name(name) => {
+                format!("`{deprecated}` is removed in Airflow 3.0; use {name} instead")
+            }
+        }
+    }
+}
+
+/// AIR302
+pub(crate) fn deprecated_members(checker: &mut Checker, expr: &Expr) {
+    let Some((deprecated, replacement)) =
+        checker
```

Comment:
We could consider adding a `Modules::Airflow` to avoid semantic name lookups for users not using airflow. See https://github.com/astral-sh/ruff/blob/928ffd66503759679823c5f346941ef58486766e/crates/ruff_linter/src/rules/numpy/rules/numpy_2_0_deprecation.rs#L162-L164

Reply:
Done in #14581 instead.